filesystem/src/main/java/com/nytimes/android/external/fs/filesystem/BreadthFirstFileTreeIterator.java

```diff
@@ -51,7 +52,9 @@
      * @param root The root directory
      */
     BreadthFirstFileTreeIterator(@Nonnull File root) {
-        this.currentList = root.listFiles();
+        if (root.listFiles() != null) {
```

Comment:
Better keep this in a single variable, in case it is a quite huge dir and it is expensive to list the files each time?

```java
File[] listedFiles = root.listFiles();
if(listedFiles != null){
    currentList = Arrays.copy(listedFiles, listedFiles.length);
}
```